CMakePresets.json

```diff
@@ -0,0 +1,35 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "common",
+            "displayName": "Common",
+            "description": "Common configuration regardless of environment or platform",
+            "cacheVariables": {
+                "BATTERY": "ON",
+                "BROADCAST": "ON",
+                "DOWNLOAD_MANUAL": "ON",
+                "HID": "ON",
+                "KEYFINDER": "ON",
+                "LILV": "ON",
+                "OPUS": "ON",
+                "QTKEYCHAIN": "ON",
+                "VINYLCONTROL": "ON",
+                "MODPLUG": "ON",
+                "WAVPACK": "ON"
```

Comment:
We shouldn't use presets to default things that are already defaulted in the `CMakeLists.txt`. This redundancy will eventually just get out-of-sync again.

I agree that explicitness in general is good, but if we store these in a common profile anyway, I am not sure how much value this adds over simplify having the right defaults.

Reply:
Valid criticism. I like this approach because the defaults in the CMakeLists.txt aren't uniform necessarily. First of all, you have to search for it in the CMakeLists.txt and then you need to make sure that the default actually applies to your environment (since they change between platforms). This takes the environment out of the equation which make it simpler to determine the configuration of the build.

Still, once I revisit this PR, I'll double check what defaults I can remove from this preset and reevaluate how much sense presets in general still make afterwards.

Reply:
Hm yeah, but usually there's a reason for some default being environment-dependent. If we can just override it with a constant, why not make it the default in the first place?

Reply:
I also think there's value in having CI produce a build that matches what a user would build by default (unless there's a strong reason to override/customize some flag), also to have a higher signal-to-noise ratio in the build configuration. That effectively means building with as few flags as possible. But I can also see why one would do otherwise for explicitness.